### PR TITLE
[V2] bug: failed operation should not be requeued if operation queue is scheduled for deletion

### PR DESCRIPTION
**Pull request by sunpa93**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
* With current implementation, failed operation can end up getting requeued to a "to be deleted" operation queue if operation queue is cleaned up in the middle of the operation.
* This can lead to situations where an operation queue for a deleted volume continues rolling with failure and flooding logs and wasting resources.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by sunpa93** [comment]
@edreed

File: pkg/controller/azvolume.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *diskv1b
 	}
 
 	// override volume operation queue to prevent any other replica operation from being executed
-	release := r.controllerSharedState.overrideAndClearOperationQueue(azVolume.Name)
+	release := r.controllerSharedState.cleanOperationQueue(azVolume.Name)
```
Comment (edreed) [inline pkg/controller/azvolume.go:204]: `closeOperationQueue` would better communicate intended use.

**Pull request merged**